crates/ruma-events/src/call/matrix_rtc.rs

```diff
@@ -0,0 +1,409 @@
+            })],
+            membership_id: "0".to_owned(),
+            created_ts: None,
+        }])
+    }
+
+    fn create_call_member_event_content() -> MemberEventContent {
+        MemberEventContent::new(
+            Application::Call(CallApplicationContent {
+                call_id: "123456".to_owned(),
+                scope: CallScope::Room,
+            }),
+            "ABCDE".to_owned(),
+            ActiveFocus::Livekit(ActiveLivekitFocus {
+                focus_select: super::focus::FocusSelection::OldestMembership,
+            }),
+            vec![Focus::Livekit(LivekitFocus {
+                alias: "1".to_owned(),
+                service_url: "https://livekit.com".to_owned(),
+            })],
+            None,
+        )
+    }
+
+    #[test]
+    fn serialize_call_member_event_content() {
+        let call_member_event = &json!(
+                {
+                    "application": "m.call",
+                    "call_id": "123456",
+                    "scope": "m.room",
+                    "device_id": "ABCDE",
+                    "foci_preferred": [
+                        {
+                            "livekit_alias": "1",
+                            "livekit_service_url": "https://livekit.com",
+                            "type": "livekit"
+                        }
+                    ],
+                    "focus_active":{
+                        "type":"livekit",
+                        "focus_select":"oldest_membership"
+                    }
+                }
+        );
+        assert_eq!(
+            call_member_event,
+            &serde_json::to_value(create_call_member_event_content()).unwrap()
+        );
+
+        let empty_call_member_event = &json!({});
+        assert_eq!(
+            empty_call_member_event,
+            &serde_json::to_value(MemberEventContent::Empty {}).unwrap()
+        );
+    }
+
+    #[test]
+    fn serialize_legacy_call_member_event_content() {
+        let call_member_event = &json!({
+            "memberships": [
+                {
+                    "application": "m.call",
+                    "call_id": "123456",
+                    "scope": "m.room",
+                    "device_id": "ABCDE",
+                    "expires": 3_600_000,
+                    "foci_active": [
+                        {
+                            "livekit_alias": "1",
+                            "livekit_service_url": "https://livekit.com",
+                            "type": "livekit"
+                        }
+                    ],
+                    "membershipID": "0"
+                }
+            ]
+        });
+
+        assert_eq!(
+            call_member_event,
+            &serde_json::to_value(create_call_member_legacy_event_content()).unwrap()
+        );
+    }
+    #[test]
+    fn deserialize_call_member_event_content() {
+        let call_member_ev: MemberEventContent = MemberEventContent::new(
```

Comment:
Specifying the type is unnecessary given that it uses a constructor of the type.

```suggestion
        let call_member_ev = MemberEventContent::new(
```